presto-main/src/main/java/com/facebook/presto/operator/LookupJoinOperator.java

```diff
@@ -35,17 +36,20 @@
     private static final int MAX_POSITIONS_EVALUATED_PER_CALL = 10000;
 
     private final OperatorContext operatorContext;
-    private final List<Type> types;
+    private final List<Type> probeTypes;
+    private final List<Type> buildTypes;
     private final ListenableFuture<? extends LookupSource> lookupSourceFuture;
     private final JoinProbeFactory joinProbeFactory;
     private final Runnable onClose;
 
-    private final PageBuilder pageBuilder;
+    private IntVector probeIndexBuilder = new IntVector();
+    private final PageBuilder buildPageBuilder;
```

Comment:
I think "lookupSource" should not be abbreviated to "source". In fact in plan join has simply two sources, so this term is ambiguous.

The build/probe sides naming is not at all self explanatory, but it seems established. Call it a jargon (although not specifically ours, https://en.wikipedia.org/wiki/Hash_join#Classic_hash_join).

I agree "build" alone (or with an addition of "PageBuilder" which doesn't explain much) is not clear, too overloaded. That's why i suggested longer name, with "output" and with "side" -- as we are page-building only one side of the join.